lib/srv/discovery/fetchers/db/azure_postgres_flex.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresqlflexibleservers"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud/azure"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/discovery/common"
+)
+
+// newAzurePostgresFlexServerFetcher creates a fetcher for Azure PostgreSQL Flexible server.
+func newAzurePostgresFlexServerFetcher(config azureFetcherConfig) (common.Fetcher, error) {
+	return newAzureFetcher[*armpostgresqlflexibleservers.Server, azure.PostgresFlexServersClient](config, &azurePostgresFlexServerFetcher{})
+}
+
+// newAzurePostgresFlexServerFetcher implements azureFetcherPlugin for Azure PostgreSQL Flexible server.
+type azurePostgresFlexServerFetcher struct {
+}
+
+// GetListClient returns a server-listing client for Azure PostgreSQL Flexible server.
+func (f *azurePostgresFlexServerFetcher) GetListClient(cfg *azureFetcherConfig, subID string) (azure.PostgresFlexServersClient, error) {
+	client, err := cfg.AzureClients.GetAzurePostgresFlexServersClient(subID)
+	return client, trace.Wrap(err)
+}
+
+// GetServerLocation returns the location of an Azure PostgreSQL Flexible server.
+func (f *azurePostgresFlexServerFetcher) GetServerLocation(server *armpostgresqlflexibleservers.Server) string {
+	return azure.StringVal(server.Location)
+}
+
+// NewDatabaseFromServer converts an Azure PostgreSQL server to a Teleport database.
+func (f *azurePostgresFlexServerFetcher) NewDatabaseFromServer(server *armpostgresqlflexibleservers.Server, log logrus.FieldLogger) types.Database {
```

Comment:
```
$ az mysql flexible-server parameter list -g database-access -s teleport-mysql-flex                                                    (base) 
[
  {
    "allowedValues": "ON,OFF",
    "dataType": "Enumeration",
    "defaultValue": "OFF",
    "description": "Setting this to ON sets the authentication to Azure AD authentication only and disable MySQL Authentication. Configure this parameter only when Azure AD authentication is configured.",
    "id": "/subscriptions/*snip*/resourceGroups/database-access/providers/Microsoft.DBforMySQL/flexibleServers/teleport-mysql-flex/configurations/aad_auth_only",
    "isConfigPendingRestart": "False",
    "isDynamicConfig": "True",
    "isReadOnly": "False",
    "name": "aad_auth_only",
    "resourceGroup": "database-access",
    "source": "user-override",
    "systemData": null,
    "type": "Microsoft.DBforMySQL/flexibleServers/configurations",
    "value": "OFF"
  },
...a lot more lines...
```

Looks like that corresponds to this SDK pager: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/mysql/armmysqlflexibleservers#ConfigurationsClient.NewListByServerPager

What's unfortunate, is that the value is set to "OFF" because I have mysql AAD auth and MySQL authentication both enabled 🥲 . I don't see another configuration setting that tells us if both are enabled.

Reply:
I couldn't find any relevant configuration for PostgreSQL either. Though the following cli shows something
```
 az postgres flexible-server ad-admin list -g database-access -s teleport-postgres-flex --debug
```
Debug flag shows something like this `'https://management.azure.com/subscriptions/<sub-id>/resourceGroups/database-access/providers/Microsoft.DBforPostgreSQL/flexibleServers/teleport-postgres-flex/administrators?api-version=2022-03-08-preview'

Reply:
`IsSupported` for single-server was to cover the case where the db was MySQL 5.6, which isn't supported for AAD auth. I think it's time to draft an RFD for Azure permissions bootstrapping and AD admin setup though - that would solve this problem comprehensively.

We had discussed this in https://github.com/gravitational/teleport/issues/14688 before.

I'll draft an RFD this week

Reply:
(to clarify) All mysql/postgres flexi server versions support AAD auth, which is why I did not include an `IsSupported` check in this PR.

Reply:
resolving this since I see no way to do it currently.

I think as everything moves into the `discovery_service` section, we can add Azure permissions & AAD Auth checking into that, maybe similar to how we use discovery service to install ssh agents.

Reply:
> All mysql/postgres flexi server versions support AAD auth, 

Yes, it's always supported but may not be enabled. We are not very consistent on this topic. For example, for RDS we even go **modify** the instance/cluster to enable IAM auth but some customers don't like us having the modify permission.

I guess we can just make it as a pre-req in the guide that AAD auth must be enabled. And I think we already have a good error message in this case.